### PR TITLE
[Feat] Attendance on Holiday - Creates Additional Salary and Compensatory Leave Allocation

### PR DESCRIPTION
**Pull request by pjamsheer**
## Feature description
Attendance on Holiday - Creates Additional Salary and Compensatory Leave Allocation

## Solution description
* **Create and Submit Additional Salary On attendance Submit and Cancel Additional Salary On attendance Cancel - Additional Salary will comes in Payroll and Salary Slip** 
* **Create and Submit Compensatory Leave Request On attendance Submit and Cancel Compensatory Leave Request On attendance Cancel - Compensatory Leave Request submit and cancel will update leave balance.**
* **If an employee is working in Site that has a Site Allowance predefined, then the public holiday modifier does not apply to their overtime because the Site Allowance is supposed to cover those public holidays, it will create Compensatory Leave Request**

## Areas affected and ensured
In `one_fm/hooks.py` added `doc_event` `on_submit` for dotype Attendance - `on_submit` of Attendance hook was having method to the path `one_fm.api....` , that moved to `one_fm.one_fm.utils.attendance_on_submit`

## Is there any existing behavior change of other features due to this code change?
No

## Was this feature tested on all the browsers?
  - [x] Chrome
  - [x] Safari

### CODE REVIEW

File: one_fm/one_fm/utils.py [hunk 1]
```diff
@@ -266,4 +267,132 @@ def notify_operator_with_supervisor_response(name):
...
+    '''
+    # Get Employee Schedule for this employee and date
+    employee_schedule = frappe.db.exists('Employee Schedule',
+        {'employee': employee, 'date': date, 'employee_availability': 'Working'})
+    if employee_schedule:
+        # Get Operations Site from the Employee Schedule
+        operations_site = frappe.db.get_value('Employee Schedule', employee_schedule, 'site')
+        if operations_site:
+            # Return include site allowance, then return true
+            return frappe.db.get_value('Operations Site', operations_site, 'include_site_allowance')
```
Comment (Amna-Alshawa) [inline one_fm/one_fm/utils.py:328]: This will return the value (include_site_allowance). It won't return "True"!!

Comment (pjamsheer) [reply one_fm/one_fm/utils.py:328]: @Amna-onefm `include_site_allowance` is a check field in Operations Site, It will give a value 0 or 1 (False or True)

Comment (Amna-Alshawa) [reply one_fm/one_fm/utils.py:328]: @pjamsheer , Oh Okay, I didn't check that.

**Review by Amna-Alshawa — APPROVED** [review]
@pjamsheer , Perfect and very organized!

**Comment by taqi-onefm** [comment]
This PR should be towards staging branch, not master-13

**Pull request closed without merging**